x/gov/handler.go

```diff
@@ -30,44 +31,23 @@ func RegisterQuery(qr weave.QueryRouter) {
 }
 
 // RegisterRoutes registers handlers for governance message processing.
-func RegisterRoutes(r weave.Registry, auth x.Authenticator) {
-	propBucket := NewProposalBucket()
-	elecBucket := NewElectorateBucket()
+func RegisterRoutes(r weave.Registry, auth x.Authenticator, decoder OptionDecoder, executor Executor) {
+	r = migration.SchemaMigratingRegistry(packageName, r)
+	r.Handle(pathVoteMsg, NewVoteHandler(auth))
+	r.Handle(pathTallyMsg, NewTallyHandler(auth, decoder, executor))
+	r.Handle(pathCreateProposalMsg, NewCreateProposalHandler(auth, decoder))
+	r.Handle(pathDeleteProposalMsg, NewDeleteProposalHandler(auth))
+	r.Handle(pathUpdateElectorateMsg, NewUpdateElectorateHandler(auth))
+	r.Handle(pathUpdateElectionRulesMsg, NewUpdateElectionRuleHandler(auth))
```

Comment:
I am not so sure in general.
In fact, when I want to make a governance router for other packages, I may only want to include one of the handlers. And thus would need to expose it directly.

However, since I expose the governance registration here, I think it is not needed in this case. The BNSd app will need to do custom registrations for eg. distribution

Reply:
I did make requested adjustments in this package though